packages/esm-appointments-app/src/home-appointments/appointments-list.component.tsx

```diff
@@ -0,0 +1,251 @@
+import { spaBasePath } from '../constants';
+
+import { ActionsMenu } from './appointment-actions.component';
+
+interface AppointmentsProps {
+  appointments: Array<MappedAppointment>;
+  isLoading: Boolean;
+  tableHeading: String;
+  mutate?: () => void;
+}
+
+const ServiceColor = ({ color }) => <div className={styles.serviceColor} style={{ backgroundColor: `${color}` }} />;
+
+const AppointmentsBaseTable: React.FC<AppointmentsProps> = () => {
+  const { appointmentsEnvironment } = useConfig();
+  const { isLoading, appointments } = useTodayAppointments();
+
+  const { t } = useTranslation();
+  const layout = useLayoutType();
+
+  const tableHeaders = useMemo(
+    () => [
+      {
+        id: 0,
+        header: t('dateTime', 'Date and time'),
+        key: 'dateTime',
+      },
+      {
+        id: 1,
+        header: t('name', 'Name / HSU'),
+        key: 'name',
+      },
+      {
+        id: 2,
+        header: t('location', 'Location'),
+        key: 'location',
+      },
+      {
+        id: 3,
+        header: t('service', 'Service'),
+        key: 'service',
+      },
+      {
+        id: 4,
+        header: 'Actions',
+        key: 'actionButton',
+      },
+    ],
+    [t],
+  );
+
+  const tableRows = appointments?.map((appointment) => ({
+    id: appointment.id,
+    dateTime: {
+      content: (
+        <span className={styles.statusContainer}>
+          <span className={styles.startTime}>{appointment.dateTime}</span>
+          {appointment.duration}
+        </span>
+      ),
+    },
+    name: {
+      content: (
+        <div className={styles.nameContainer}>
+          <ConfigurableLink to={`\${openmrsSpaBase}/patient/${appointment.patientUuid}/chart`}>
+            {appointment.name}
+          </ConfigurableLink>
+          <span className={styles.identfier}>{appointment.identifier}</span>
+        </div>
+      ),
+    },
+    location: {
+      content: <span className={styles.statusContainer}>{appointment.location}</span>,
+    },
+    service: {
+      content: (
+        <span className={styles.serviceContainer}>
+          <ServiceColor color={appointment.serviceColor} />
+          {appointment.serviceType}
+        </span>
+      ),
+    },
+    actionButton: {
+      content: (
+        <span className={styles.serviceContainer}>
+          {appointment.status === 'Completed' ? (
+            <div className={styles.completeIcon}>
+              <WatsonHealthStatusResolved /> Completed
+            </div>
+          ) : appointment.status === 'CheckedIn' ? (
+            <Button kind="primary" className={styles.actionButton}>
+              COMPLETED
+            </Button>
+          ) : (
+            <Button kind="primary" className={styles.actionButton}>
+              CHECK IN
+            </Button>
+          )}
+          <ActionsMenu appointment={appointment} environment={appointmentsEnvironment} />
+        </span>
+      ),
+    },
+  }));
+
+  const AddAppointmentLink = () =>
```

Comment:
This component has no implicit props and could be moved outside of `AppointmentsBaseTable` to help memoization.